generator/src/generator.ts

```diff
@@ -1669,6 +1669,47 @@ export class Generator {
     if (this.combineSubpaths) map.combineSubpaths();
     return map.toJSON();
   }
+
+  /**
+   *
+   * @param outputDirectory The output directory, resolved relative to process.cwd(), where the import
+   * map and its files are stored on the file system. Defaults to 'dist'
+   * @param importMapFileName The name of the import map file stored to disk, within the outputDirectory. Defaults to
+   * main.importmap
+   * @param hostedBaseUrl a fully qualified or relative string path used as base url for all URLs within the import
+   * If not provided, hostedBaseUrl defaults to './'
+   */
+  async downloadAll(
+    outputDirectory: string = 'deps',
+    importMapFileName: string = 'main.importmap',
```

Comment:
I'd prefer if we could avoid this property. Generator operations are all import map in, import map out. So I think this one can follow the same process and not explicitly do the import map save but assume it is wrapped in a `inputMap` input and `getImportMap()` output. Better clarifying this "input / output" design of the generator would be nice, but it applies across all commands!

Reply:
I'm happy to remove `importMapFileName` and saving the import map to disk. What do you mean by "import map in, import map out?" I see that some methods call `this.importMap.clone()` rather than modifying `this.importMap` - is that related? Within `downloadAll`, I think the only change to the import map is to rebase the URL away from the registry urls to ones relative to `hostedBaseUrl`. Perhaps `downloadAll` should return that cloned, modified import map?

Reply:
Rather than returning the cloned and modified import map, the current pattern in the generator is to actually mutate `generator.importMap`. Think of `generaator.importMap` as an exact mirror of the state of `importmap.json`. The JSPM CLI then wraps all generator calls with picking up the JSON file and saving it back when using a command.

Are there any reasons mutation can't work for your use case here?